src/ai/backend/storage/api/manager.py

```diff
@@ -776,15 +779,53 @@ class Params(TypedDict):
     ) as params:
         await log_manager_api_entry(log, "mkdir", params)
         ctx: RootContext = request.app["ctx"]
+        vfid = params["vfid"]
+        parents = params["parents"]
+        exist_ok = params["exist_ok"]
+        relpath = params["relpath"]
+        relpaths = relpath if isinstance(relpath, list) else [relpath]
+        failure_tasks: list[str] = []
+        success_tasks: list[str] = []
+
         async with ctx.get_volume(params["volume"]) as volume:
-            with handle_fs_errors(volume, params["vfid"]):
-                await volume.mkdir(
-                    params["vfid"],
-                    params["relpath"],
-                    parents=params["parents"],
-                    exist_ok=params["exist_ok"],
+            mkdir_tasks = [
+                volume.mkdir(vfid, rpath, parents=parents, exist_ok=exist_ok) for rpath in relpaths
+            ]
+            result_group = await asyncio.gather(*mkdir_tasks, return_exceptions=True)
+            exception_flags = [isinstance(res, BaseException) for res in result_group]
+
+            # if result group is all false throw error.
+            if all(exception_flags):
+                raise web.HTTPInternalServerError(
+                    body=json.dumps(
+                        {
+                            "msg": "FileExistsError None of directories created!",
```

Comment:
How are you sure that all exceptions are `FileExistsError`?
We should not "assume" any exception but always explicitly report the original information whenever possible.

In this case, I think we could sample the first exception instance, so that the user may retry the operation with revised inputs, though he/she may experience other remaining errors.